internal/worker/objectstore/worker.go

```diff
@@ -192,6 +202,28 @@ func (w *objectStoreWorker) Wait() error {
 	return w.catacomb.Wait()
 }
 
+// FlushWorkers flushes the object store workers.
+func (w *objectStoreWorker) FlushWorkers(ctx context.Context) error {
+	// We have to synchronise the flush workers to ensure that we don't
+	// have multiple flushes happening at the same time and that we aren't
+	// creating new workers whilst flushing.
+	select {
+	case <-w.catacomb.Dying():
+		return w.catacomb.ErrDying()
+	case w.flushWorkers <- struct{}{}:
+	}
+	return nil
+}
+
+func (w *objectStoreWorker) stopAndRemoveAllWorkers(ctx context.Context) error {
+	for _, namespace := range w.runner.WorkerNames() {
+		if err := w.runner.StopAndRemoveWorker(namespace, ctx.Done()); err != nil && !errors.Is(err, errors.NotFound) {
+			return errors.Trace(err)
+		}
```

Comment:
Are we on a line budget?
```suggestion
		err := w.runner.StopAndRemoveWorker(namespace, ctx.Done())
		if err != nil && !errors.Is(err, errors.NotFound) {
			return errors.Trace(err)
		}
```